core/templates/pages/topic-viewer-page/practice-tab/practice-tab.component.ts

```diff
@@ -95,6 +105,45 @@ export class PracticeTabComponent implements OnInit {
       this.classroomUrlFragment = (
         this.urlService.getClassroomUrlFragmentFromLearnerUrl());
     }
+    this.topicViewerBackendApiService.fetchTopicDataAsync(
+      this.topicUrlFragment, this.classroomUrlFragment).then(
```

Comment:
So, the whole topic is fetched just for its ID? If so, can’t we get the ID from some other service that already has that data?

Reply:
I've modified the component so that the `topicId` is received as input from its parent component instead. Done.